packages/wallet/primitives/src/session-signature.ts

```diff
@@ -117,8 +128,23 @@ export function encodeSessionCallSignatures(
     throw new Error('Incomplete topology')
   }
 
+  if (implicitSigners.length > 0 && !identitySigner) {
+    console.warn('Encodings using implicit signers without identity signer may encode incorrectly')
+  }
+
+  // Check the topology contains exactly one identity signer
+  const identitySigners = getIdentitySigners(topology)
+  if (identitySigner) {
+    if (!identitySigners.some((s) => Address.isEqual(s, identitySigner!))) {
+      throw new Error('Identity signer not found')
+    }
+  } else {
+    // Grab the first one
+    identitySigner = identitySigners[0]!
```

Comment:
I don't think this is a good default, right? it may mask errors working in one-off scenarios

Reply:
When using explicit signers only it doesn't matter which identity signer we use. We can make this required here but further up the chain we still need to pull a random identity signer for when we aren't using implicits.

Reply:
Updated